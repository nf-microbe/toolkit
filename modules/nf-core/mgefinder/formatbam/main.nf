process MGEFINDER_FORMATBAM {
    tag "${meta.id}"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://carsonjm/mgefinder:0.0.1' :
        'docker://carsonjm/mgefinder:0.0.1' }"

    input:
    tuple val(meta), path(bam)

    output:
    tuple val(meta), path("${prefix}.mgefinder.bam*")   , emit: bam
    path "versions.yml"                                 , emit: versions

    script:
    def args = task.ext.args   ?: ''
    prefix   = task.ext.prefix ?: "${meta.id}"
    """
    mgefinder \\
        formatbam \\
        ${bam} \\
        ${prefix}.mgefinder.bam \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgefinder: \$(echo \$( mgefinder --version | tail -n 1 | sed "s/MGEfinder version://" ))
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.mgefinder.bam
    touch ${prefix}.mgefinder.bam.bai

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgefinder: \$(echo \$( mgefinder --version | tail -n 1 | sed "s/MGEfinder version://" ))
    END_VERSIONS
    """
}
