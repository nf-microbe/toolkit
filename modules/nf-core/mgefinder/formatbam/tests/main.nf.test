nextflow_process {

    name "Test Process MGEFINDER_FORMATBAM"
    script "../main.nf"
    process "MGEFINDER_FORMATBAM"

    tag "modules"
    tag "modules_nfcore"
    tag "mgefinder"
    tag "mgefinder/formatbam"

    test("bam") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file("/mmfs1/gscratch/pedslabs_hoffman/carsonjm/nf-microbe/modules/.nf-test/tests/d66d38376a064f69b0a249d9c75d0d88/work/62/0a9b7e06c284d93c10996a4881b416/test.bam", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("bam - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    []
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
