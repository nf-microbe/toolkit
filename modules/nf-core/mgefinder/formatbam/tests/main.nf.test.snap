{
    "bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test.mgefinder.bam:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test.mgefinder.bam.bai:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,68fb8eb0a87c3128ae275e2103e176f1"
                ],
                "bam": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test.mgefinder.bam:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test.mgefinder.bam.bai:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,68fb8eb0a87c3128ae275e2103e176f1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-07T21:58:00.88243674"
    },
    "bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test.mgefinder.bam:md5,6eacef59dd393c714a02075c0d6e59cd",
                            "test.mgefinder.bam.bai:md5,48771f50863f34fd977780f508893388"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,68fb8eb0a87c3128ae275e2103e176f1"
                ],
                "bam": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test.mgefinder.bam:md5,6eacef59dd393c714a02075c0d6e59cd",
                            "test.mgefinder.bam.bai:md5,48771f50863f34fd977780f508893388"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,68fb8eb0a87c3128ae275e2103e176f1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-07T16:24:20.153906531"
    }
}